rtl/ip/system_info/util/system_info_gen.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3.10
+# Copyright lowRISC contributors
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+from pathlib import Path
+from typing import NamedTuple
+
+import git
+from mako.template import Template
+
+
+class SystemInfo(NamedTuple):
+    commit_hash: str
+    dirty: bool
+
+
+def generate_system_info() -> None:
+    try:
+        repo = git.Repo(search_parent_directories=True)
+        commit = repo.head.object.hexsha
+        dirty = repo.is_dirty()
+    except git.exc.InvalidGitRepositoryError:
+        commit = os.environ.get("FLAKE_GIT_COMMIT")
+        dirty = bool(os.environ.get("FLAKE_GIT_DIRTY"))
+        if commit is None or dirty is None:
+            print("Error: Couldn't get commit and dirty via git or Nix")
+            exit()
```

Comment:
It's probably a little cleaner to reverse the order of the check (check if the FLAKE_* envvars exist first), and then `import git # noqa: E402` inside the else path. Then you won't need to add pkgs.git dependency for the nix builds (sonata-simulator{-lint}).

Reply:
Ah interesting I hadn't thought of doing the input inside the if statement. I do find it weird that it requires the git executable just for importing the library. Should now be fixed.
Ruff check removed the `noqa` comment, I hope that's ok.